### PR TITLE
changing to node 14

### PR DESCRIPTION
**Pull request by sdh100shaun**
## Description

Updates starter to node 14 - closes https://github.com/dvsa/dvsa-lambda-starter/issues/43
Also adds security workflow to the pull requests, using Snyk

Related issue: [https://github.com/dvsa/dvsa-lambda-starter/issues/43] 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Hodgy** [comment]
Do we need the empty `snyk.sarif` file? Is this not generated by a step in the snyk action and then uploaded?

File: .github/workflows/security.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        continue-on-error: true # To make sure that SARIF upload gets called
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          args: --sarif-file-output=snyk.sarif
+          json: true
+      - name: Upload result to GitHub Code Scanning
+        uses: github/codeql-action/upload-sarif@v1
+        with:
+          sarif_file: snyk.sarif
```
Comment (jerotire) [inline .github/workflows/security.yaml:19]: Minor: No new line at end of file.

**Review by jerotire — CHANGES_REQUESTED** [review]
> Do we need the empty `snyk.sarif` file? Is this not generated by a step in the snyk action and then uploaded?
Concur, that's the output file for snyk runs right? So it should not be source controlled, as that is generated during the workflow/actions and imported/consumed by `github/codeql-action/upload-sarif@v1`, and workspace discarded?

File: .github/workflows/security.yaml (hunk 1 above)
Comment (sdh100shaun) [reply .github/workflows/security.yaml:19]: I put the file there cos it was not running at that time will try removing it

Comment (sdh100shaun) [reply .github/workflows/security.yaml:19]: file removed all still works

**Review by jerotire — APPROVED** [review]

**Review by sjduffey — APPROVED** [review]

**Pull request merged**